htdocs/core/class/commonobject.class.php

```diff
@@ -5410,6 +5410,11 @@ protected function commonGenerateDocument($modelspath, $modele, $outputlangs, $h
 				// output format that does not support UTF8.
 				$sav_charset_output = $outputlangs->charset_output;
 
+				// memorize model name for pdf hooks (ie pdf_writelinedesc when validating a shipping)
+				if (empty($this->model_pdf)) {
```

Comment:
hello @eldy i did not understand your question. 
This small PR only helps to transmit model name to hooks like pdf_writelinedesc as said in comment. I dont see why it will cause any trouble since it just affects the current instance of the object and is not updated in database.
I must miss something.

Reply:
Why do you have the test 
If (empty($this->model))

Reply:
@eldy you are right ! i made another PR to siplify my work (conflicts) : https://github.com/Dolibarr/dolibarr/pull/25918